core/vm/environment.go

```diff
@@ -35,6 +37,7 @@ type (
 
 // Context provides the EVM with auxilary information. Once provided it shouldn't be modified.
 type Context struct {
+	context.Context
```

Comment:
Yes, I'm aware of that. But circumstances are different here as well since the context is fully optional. Please provide an alternative approach.